src/esp/assets/ResourceManager.cpp

```diff
@@ -2961,15 +2961,27 @@ void ResourceManager::loadTextures(Importer& importer,
         if (image->isCompressed()) {
           format = Mn::GL::textureFormat(image->compressedFormat());
         } else {
-          format = Mn::GL::textureFormat(image->format());
+          const auto pixelFormat = image->format();
+          format = Mn::GL::textureFormat(pixelFormat);
           // Modify swizzle for single channel textures so that they are
           // greyscale
-          if (pixelFormatSize(image->format()) == 1) {
+          if (pixelFormatChannelCount(pixelFormat) == 1) {
```

Comment:
The pixel format property queries are relatively large switches internally, so you may want to cache the result instead of querying twice.